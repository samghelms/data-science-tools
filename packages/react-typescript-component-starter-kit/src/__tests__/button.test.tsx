import * as React from "react";
import renderer from "react-test-renderer";
import { Button } from "../";
import { mount } from "enzyme";
import sinon from "sinon";

describe("component: Button", () => {
  test("snapshot", () => {
    const button = renderer.create(
      <Button onClick={() => {}} kind="primary" outline>
        Click me
      </Button>
    );

    const tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("onClick should trigger correctly", () => {
    const spy = sinon.spy();
    const button = mount(
      <Button onClick={spy} kind="primary" outline>
        Click me
      </Button>
    );

    button.simulate("click");
    expect(spy.calledOnce).toBeTruthy();
  });
});
