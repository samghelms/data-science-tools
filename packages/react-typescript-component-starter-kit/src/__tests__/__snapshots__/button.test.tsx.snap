// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Button snapshot 1`] = `
.emotion-0 {
  cursor: pointer;
  margin: 3px 5px;
  border: none;
  border-radius: 3px;
}

<button
  className="emotion-0"
  onClick={[Function]}
  style={
    Object {
      "&:hover": Object {
        "boxShadow": "inset 0 0 0 1000px #1FB6FF",
        "color": "white",
      },
      "background": "transparent",
      "boxshadow": "inset 0 0 0 1px #1FB6FF",
      "color": "#1FB6FF",
      "transition": "all .3s",
    }
  }
>
  Click me
</button>
`;
